### PR TITLE
Updates building instruction

### PR DESCRIPTION
**Pull request by crsib**
* Added Troubleshooting Conan section
* Updated tools version

Resolves: #2102

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -206,6 +206,32 @@ This option implies `-Daudacity_obey_system_dependencies=On` and disables `local
...
 
+### Troubleshooting Conan issues
+
+To fix errors, similar to the following:
+
+```
+ERROR: HTTPSConnectionPool(host='center.conan.io', port=443): Max retries exceeded with url: /v1/ping (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1131)
+```
+
+please, update you Conan version. Alternatively, the problem can be fixed using:
```
Comment (Paul-Licameli) [inline BUILDING.md:217]: "your"

File: BUILDING.md [hunk 2]
```diff
@@ -206,6 +206,32 @@ This option implies `-Daudacity_obey_system_dependencies=On` and disables `local
...
+
+```
+$ conan config install https://github.com/conan-io/conanclientcert.git
+```
+
+For the errors like:
```
Comment (Paul-Licameli) [inline BUILDING.md:223]: "For errors like:"

File: BUILDING.md [hunk 3]
```diff
@@ -206,6 +206,32 @@ This option implies `-Daudacity_obey_system_dependencies=On` and disables `local
...
+
+```
+Unable to connect to conan-center=https://conan.bintray.com
+```
+
+please, run
```
Comment (Paul-Licameli) [inline BUILDING.md:229]: "please run"

**Comment by Paul-Licameli** [comment]
Just some spelling and grammar suggestions.

Was I supposed to get XCode 13?   Funny I'm still on 12.5 :-)

**Comment by crsib** [comment]
> Was I supposed to get XCode 13? Funny I'm still on 12.5 :-)

Hmm, you do not use XCode from AppStore? I believe XCode 13 is available for macOS 11

**Comment by crsib** [comment]
> Just some spelling and grammar suggestions.

Fixed now!

**Review by Paul-Licameli — APPROVED** [review]

**Pull request merged**